crates/tbdex/src/messages/cancel.rs

```diff
@@ -0,0 +1,105 @@
+use super::{MessageKind, MessageMetadata, Result};
+use crate::json_schemas::generated::{CANCEL_DATA_JSON_SCHEMA, MESSAGE_JSON_SCHEMA};
+use chrono::Utc;
+use serde::{Deserialize, Serialize};
+use web5::dids::bearer_did::BearerDid;
+
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq)]
+pub struct Cancel {
+    pub metadata: MessageMetadata,
+    pub data: CancelData,
+    pub signature: String,
+}
+
+impl Cancel {
+    pub fn new(
+        bearer_did: &BearerDid,
+        to: &str,
+        from: &str,
+        exchange_id: &str,
+        data: &CancelData,
+        protocol: &str,
+        external_id: Option<String>,
+    ) -> Result<Self> {
+        let metadata = MessageMetadata {
+            from: from.to_string(),
+            to: to.to_string(),
+            kind: MessageKind::Cancel,
+            id: MessageKind::Cancel.typesafe_id()?,
+            exchange_id: exchange_id.to_string(),
+            external_id,
+            protocol: protocol.to_string(),
+            created_at: Utc::now().to_rfc3339(),
```

Comment:
have we double checked that we can round trip time to and from rs <-> kt without losing precision ? I know there was a nano second problem at one point

Reply:
it's a great question, we should run a full audit at some point soon. I did run into that issue a month or so ago and remediated it, but I just solved for the specific problem I was up against (I think it was VC stuff in web5 IIRC)